dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/query/JpaCriteriaQueryEngine.java

```diff
@@ -215,36 +183,26 @@ public long count(Query query) {
     criteriaQuery.select(builder.count(root));
 
     Predicate predicate = buildPredicates(builder, root, query);
-
-    boolean shareable = schema.isShareable();
-    if (shareable && !query.isSkipSharing()) {
-      UserDetails currentUserDetails = query.getCurrentUserDetails();
-      predicate
-          .getExpressions()
-          .addAll(
-              store.getSharingPredicates(builder, currentUserDetails).stream()
-                  .map(t -> t.apply(root))
-                  .toList());
-    }
-
+    addSharingPredicates(query, schema, predicate, store, builder, root);
     criteriaQuery.where(predicate);
 
-    if (!query.getOrders().isEmpty()) {
-      criteriaQuery.orderBy(
-          query.getOrders().stream()
-              .map(
-                  o ->
-                      o.isAscending()
-                          ? builder.asc(root.get(o.getProperty().getName()))
-                          : builder.desc(root.get(o.getProperty().getName())))
```

Comment:
FYI: I think using `getName()` instead of `getFieldName()` was a bug - new code uses a common extracted method for fetch and count query